appsec/appsec.go

```diff
@@ -153,3 +154,32 @@ func getRootSpan(ctx context.Context) tracer.Span {
 	log.Error("appsec: could not access the root span")
 	return nil
 }
+
+var tracerNotStarted sync.Once
+
+// Start is the entrypoint to start Appsec in your application.
+// It starts the Appsec security protections according to the configured the security rules in DataDog UI
+// and the given options. The main way to start appsec is to set `DD_APPSEC_ENABLED=true` in your environment.
+// You can also use the `Start()` function to start appsec in your application to have a have a finer control on when to start it.
+// and when to stop it. All options that can be passed to `Start()` are documented in the `options` package and have a
+// environment variable equivalent.
+// Note: APM Tracer has to be started before calling this function.
+func Start(opts ...options.StartOption) {
+	if !tracer.IsUpAndRunning() {
+		tracerNotStarted.Do(func() { log.Error("appsec: tracer not started. appsec won't start. Please use tracer.Start() first") })
+		return
+	}
```

Comment:
WDYT of doing it the other way around: asking `public_appsec.Start` users to do it before `tracer.Start` so that we can just accept the first config being started with. Today's implementation restarts when `private_appsec.Start` is called multiple times.

```
appsec.Start(cfg1)
tracer.Start(cfg2)
```

or even simpler for this first iteration:

```
tracer.Start(tracer.WithAppSecConfig(cfg))
```